lib/Rex/CLI.pm

```diff
@@ -455,6 +454,30 @@ sub __help__ {
 
 }
 
+# Check whether Rex/Template.pm
+# or Rex/Template/NG.pm modules are loaded. If there are,
+# they should be present in the %INC hash.
+sub check_template_modules {
+  my @tpl_modules_loaded =
+    grep { /(Rex\/Template|Rex\/Template\/NG)/ } keys %INC;
+  if ( scalar @tpl_modules_loaded > 0 ) {
+    Rex::Logger::info(
+      "WARNING! You are using Rex built-in template engine aka Rex::Template.",
+      "warn"
+    );
+    Rex::Logger::info(
+      "This engine is obsolete, unmaintained and will be retired in future releases.",
+      "warn"
+    );
+    Rex::Logger::info(
+      "Please consider migrating to a more modern template engine.", "warn" );
+    Rex::Logger::info(
+      "Text::Template::Simple, Template::Toolkit, etc. As usual, CPAN is you friend.",
+      "warn"
+    );
```

Comment:
What do you think about pointing to [Rex::Template::TT](https://metacpan.org/pod/Rex::Template::TT) as a readily available and recommended external module, _and_ point to the docs about how to set one's own custom templating functions:

- [set_template_function](https://metacpan.org/pod/Rex::Config#set_template_function) config option
- [template_function](https://metacpan.org/pod/Rex::Commands#template_function) command

Reply:
I can do that indeed.